### DISCUSSION TITLE
Unable to stage filenames with particular croatian/serbian letters

### DISCUSSION THREAD

**Discussion by BioEssence** [magit/magit#4609] [Support] [answered]
Hello,

I am unable to stage files which contain letters "č", "ć" or "đ" in their filenames.

Those letters are used in croatian/serbian languages and are a part of [Gaj's Latin alphabet](https://en.wikipedia.org/wiki/Gaj%27s_Latin_alphabet).

Files which contain letters "š" or "ž" from the mentioned alphabet can be successfully staged.

When I try to stage the following files:
```
Untracked files (6)
anđeo.txt
kuća.txt
čaj.txt
šuma.txt
željezo.txt
```

I get the following output in buffer `magit-process`:
```
128 git.exe … "add" "--" "an eo.txt"
fatal: pathspec 'an eo.txt' did not match any files

128 git.exe … "add" "--" "ku a.txt"
fatal: pathspec 'ku a.txt' did not match any files

128 git.exe … "add" "--" " aj.txt"
fatal: pathspec ' aj.txt' did not match any files

  0 git.exe … "add" "--" "\232uma.txt"
  0 git.exe … "add" "--" "\236eljezo.txt"
```

It seems that the issue is similar to https://github.com/magit/magit/issues/3111.

I have tested this behavior in vanilla Emacs 27.2 with only Magit loaded.

Output of the `magit-version` command is:
`Magit 20220211.2231 [>= 3.3.0-git], Git 2.35.1.windows.2, Emacs 27.2, windows-nt`

Thanks for the help.

**Comment by tarsius** [comment]
I can't really help you much with that. You have to get the various involved things to agree on what encoding should be used.

Emacs/Magit uses unicode by default (the mode-line should begin with a `U`) and I think a good approach would be to get Windows to go along with that.

**Comment by tarsius** [comment]
I suggest asking somewhere else for help, i.e. the emacs reddit. There are bound to be more people there who have run into similar problems, including users of Windows.

**Comment by BioEssence** [comment, marked as answer]
Thanks for the suggestion.

I have done some research and found a solution. The character encoding in the Windows console had to be set to UTF-8. Now Magit successfully stages files which contain letters "č", "ć" or "đ" in their filenames.

Source:
https://stackoverflow.com/questions/57131654/using-utf-8-encoding-chcp-65001-in-command-prompt-windows-powershell-window/57134096#57134096

**Reply by tarsius** [reply]
Thanks for letting us know!
